umbral/umbral.py

```diff
@@ -388,36 +424,35 @@ def encapsulate(self, pub_key, key_length=32):
 
         return key, Capsule(point_eph_e=pub_r, point_eph_v=pub_u, bn_sig=s)
 
-
     def decapsulate_original(self, priv_key, capsule, key_length=32):
         """Derive the same symmetric key"""
 
-        shared_key = (capsule.point_eph_e + capsule.point_eph_v) * priv_key
+        shared_key = capsule.original_ephemeral_points() * priv_key
         key = kdf(shared_key, key_length)
 
         # Check correctness of original ciphertext (check nº 2) at the end 
         # to avoid timing oracles
         assert capsule.verify(self.params), "Generic Umbral Error"
         return key
 
-    def decapsulate_reencrypted(self, pub_key: Point, priv_key: BigNum, orig_pub_key: Point,
-                                recapsule: ReconstructedCapsule, original_capsule: Capsule, key_length=32):
+    def _decapsulate_reencrypted(self, pub_key: Point, priv_key: BigNum, orig_pub_key: Point,
+                                 capsule: Capsule, key_length=32):
         """Derive the same symmetric key"""
 
-        xcomp = recapsule.point_eph_ni
-        d = hash_to_bn([xcomp, pub_key, xcomp * priv_key], self.params)
-
-        e_prime = recapsule.point_eph_e_prime
-        v_prime = recapsule.point_eph_v_prime
-
-        shared_key = (e_prime + v_prime) * d
-        key = kdf(shared_key, key_length)
-
-        e = original_capsule.point_eph_e
-        v = original_capsule.point_eph_v
-        s = original_capsule.bn_sig
-        h = hash_to_bn([e, v], self.params)
-        inv_d = ~d
-        assert orig_pub_key * (s * inv_d) == v_prime + (e_prime * h), "Generic Umbral Error"
+        xcomp = capsule._point_noninteractive
 
-        return key
+        d = hash_to_bn([xcomp, pub_key, xcomp * priv_key], self.params)
+        shared_key = capsule.reconstructed_ephemeral_points() * d
```

Comment:
Don't do this, this is not clear at all with what's happening. It doesn't look like DH anymore.